scipy/signal/fir_filter_design.py

```diff
@@ -632,13 +637,24 @@ def firwin2(numtaps, freq, gain, nfreqs=None, window='hamming', nyq=None,
     if nfreqs is None:
         nfreqs = 1 + 2 ** int(ceil(log(numtaps, 2)))
 
+    d = np.diff(freq)
+    if (d == 0).any():
+        freq = np.array(freq, copy=True)
+
     # Tweak any repeated values in freq so that interp works.
-    eps = np.finfo(float).eps
+    eps = np.finfo(float).eps * nyq
     for k in range(len(freq)):
```

Comment:
It seems that this whole loop, and subsequent strict increasing check, could be housed within the `if (d == 0).any():` conditional block above, since it is only then that any `freq[k] == freq[k + 1]`.